.buildkite/pipeline.yml

```diff
@@ -43,6 +41,42 @@ steps:
           files: "lib/*/unittest/boost_test_results.junit"
           format: "junit"
   
+  - label: "Build :cpp: for linux_aarch64_cross-RelWithDebInfo"
+    timeout_in_minutes: 180
+    agents:
+      cpu: "6"
+      ephemeralStorage: "20G"
+      memory: "64G"
+      image: "docker.elastic.co/ml-dev/ml-linux-aarch64-cross-build:10"
+    commands:
+      - 'echo "checkout succeeded"'
+      - ".buildkite/scripts/steps/build_and_test.sh"
+    depends_on: "check_style"
+    key: "build_linux_aarch64_cross-RelWithDebInfo"
+    env:
+      CPP_CROSS_COMPILE: "aarch64"
+      CMAKE_FLAGS: "-DCMAKE_TOOLCHAIN_FILE=cmake/linux-aarch64.cmake"
+      RUN_TESTS: "false"
+
+  - label: "Build :cpp: for macos_aarch64_cross-RelWithDebInfo"
+    timeout_in_minutes: 180
+    agents:
+      cpu: "6"
+      ephemeralStorage: "20G"
+      memory: "64G"
+      image: "docker.elastic.co/ml-dev/ml-macosx-build:16"
+    commands:
+      - 'echo "checkout succeeded"'
+      - ".buildkite/scripts/steps/build_and_test.sh"
+    depends_on: "check_style"
+    key: "build_macos_aarch64_cross-RelWithDebInfo"
+    env:
+      CPP_CROSS_COMPILE: "macosx"
+      CMAKE_FLAGS: "-DCMAKE_TOOLCHAIN_FILE=cmake/darwin-x86_64.cmake"
+      RUN_TESTS: "false"
+
+  - wait
```

Comment:
I think it's best if the `wait` goes just before uploading artifacts at the very end. Running the Linux integration tests doesn't depend on macOS being built for example.